src/nova_tz/keyblob.rs

```diff
@@ -0,0 +1,120 @@
+use std::array::TryFromSliceError;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Can't parse key blob: {0}")]
+    BadKeyBlob(String),
+}
+
+pub enum DigestPadAlgo {
+    RsaDigestPaddingNone = 0,
+    RsaPkcs115Sha2_256 = 1,
+    RsaPssSha2_256 = 2,
+}
+
+impl TryFrom<u32> for DigestPadAlgo {
+    type Error = Error;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(DigestPadAlgo::RsaDigestPaddingNone),
+            1 => Ok(DigestPadAlgo::RsaPkcs115Sha2_256),
+            2 => Ok(DigestPadAlgo::RsaPssSha2_256),
+            _ => Err(Error::BadKeyBlob(format!(
+                "Bad digest padding algorithm {}",
+                value
+            ))),
+        }
+    }
+}
+
+pub(crate) struct KeyBlob {
```

Comment:
So technically a `KeyBlob` _is_ a `TrustZoneKeypair` or `TZKeypair` right? Why can't it be called that?